### ISSUE TITLE
[Boss/Spells] Acidmaw & Dreadscale (Trial of the Crusader)

### ISSUE THREAD

**Issue by Dranah** [WoWManiaUK/Redemption#3576]
https://www.wowhead.com/npc=35144/acidmaw
https://www.wowhead.com/npc=34799/dreadscale

Following issues I found so far about these puny worms:

1) Burning Bite not hitting Dreadscale's tank when he kites
https://www.wowhead.com/spell=66869/burning-bile often fails to be applied to Dreadscale's tank when he is mobile.

This occurs when the tank is kiting the worm out of slime pools and outranges his melee attack. The hitbox of the worm is incredibly small, he follows you rather slowly, and the ability which applies the buff https://www.wowhead.com/spell=66879/burning-bite  is melee range, therefore the worm cannot use it when you are too far away. Which is, nearly always right now.

**How it should work:** The worm's hitbox should be way larger, he should always be in melee range of the tank even when he is kiting. If the tank is somehow still outranging the attack, the worm should immediately attempt to recast it as soon as he is in range.
TL;DR burning bite should NEVER fail to cast, the dreadscale tank should have 100% uptime on burning bile. There is no precedent on retail to the bile not being applied.

2) Timers of Dreadscale
**How it should work**: https://www.wowhead.com/spell=66879/burning-bite should always be the first cast of Dreadscale when he is mobile, applying it to the tank approx 6-7 seconds into the fight, see video here: https://www.youtube.com/watch?v=F6QQIeglwY0
He should then reapply the bile every ~18-20seconds (note: the bile debuff lasts 24s), by casting Burning Bite, until he submerges. The Burning Bite should never be suppressed or delayed by any other ability. 
**How it works atm:** the initial cast of bile is missing, is only used after 20-30seconds (if doesn't fail), and is then never reapplied before the worms submerge.

3) Submerge/Emerge timers are off
**How it should work:** The first submerge and swap of the worms should occur with a bit of delay compared to the 45s timer. For this the reason is that the 45s timer should only start when the worms both fully emerged upon the pull and that's when Acid begins casting his spew, picking up an aggro target (roughly 3-4s after Dreadscale becomes attackable). Both worms should submerge at the same moment, and should emerge at the same time as well.
**How it works here:** Here the two worms arent in sync. Because dreadscale becomes attackable sooner, his submerge timer seems to begin 5s earlier than Acidmaw beginning his own timer. 
Dreadscale submerges 5s earlier and Acidmaw stays 5s longer. 
Dread re-emerges 5s earlier than Acidmaw as well. This happens on every submerge.

4) Acidmaw timers
**How it should work:** Paralytic Spray https://www.wowhead.com/spell=66901/paralytic-spray should be first cast roughly 12sec after the pull and then every ~18-20s afterwards while Acid is rooted. This means a fix amount of 2 toxin casts before the submerge, as he will burrow before the 3rd cast.
Once Acidmaw becomes mobile, his Paralytic Bite https://www.wowhead.com/spell=66824/paralytic-bite should be used immediately after 6-7seconds, and then every 25s on his tank.
**How it works here:**
Acidmaw casts the first spray only after 20-30seconds but regularly double casts it after 5-6sec again. He then casts another toxin spray after Dreadscale has already submerged while he is still up, lagging behind due to the desync'd timers.
Note: at this point your tank's bile has long expired, if he ever received it at all. You're facing at least 3 members paralyzed with no means to remove it.
Once Acidmaw becomes mobile, his Paralytic Bite is only used after 20s, the initial cast is missing just like for Burning Bite.

**Comment by Dranah** [comment]
Paralytic spray timer is now correct but still double casting.
Submerge and Paralytic Bite no change.

**Comment by Keader** [comment]
I tested myself this time.
Paralytic spray will not double cast anymore.
Submerge i try sync more possible, is not 100% but close enoght... more then that need my rewrite (good luck porting that)
Paralytic bite is there now, (i tested too)

**Comment by Dranah** [comment]
Good enough.
